src/tests/JIT/HardwareIntrinsics/Arm/Shared/SveGatherVectorFirstFaultingIndices.template

```diff
@@ -0,0 +1,633 @@
+
+            _fld2 = ({Op2BaseType}*)_dataTable.inArray2Ptr;
+        }
+
+        public bool IsSupported => {Isa}.IsSupported;
+
+        public bool Succeeded { get; set; }
+
+        public void RunBasicScenario_UnsafeRead()
+        {
+            TestLibrary.TestFramework.BeginScenario(nameof(RunBasicScenario_UnsafeRead));
+
+            var result = {Isa}.{Method}(
+                Unsafe.Read<{Op1VectorType}<{Op1BaseType}>>(_dataTable.inArray1Ptr),
+                ({Op2BaseType}*)_dataTable.inArray2Ptr,
+                Unsafe.Read<{Op3VectorType}<{Op3BaseType}>>(_dataTable.inArray3Ptr)
+            );
+
+            Unsafe.Write(_dataTable.outArrayPtr, result);
+            ValidateResult(_dataTable.inArray1Ptr, _dataTable.inArray2Ptr, _dataTable.inArray3Ptr, _dataTable.outArrayPtr);
+        }
+
+        public void RunBasicScenario_Load()
+        {
+            TestLibrary.TestFramework.BeginScenario(nameof(RunBasicScenario_Load));
+
+            {Op1VectorType}<{Op1BaseType}> loadMask1 = Sve.CreateTrueMask{Op1BaseType}(SveMaskPattern.All);
+            {Op3VectorType}<{Op3BaseType}> loadMask3 = Sve.CreateTrueMask{Op3BaseType}(SveMaskPattern.All);
+
+            var result = {Isa}.{Method}(
+                {LoadIsa}.Load{Op1VectorType}(loadMask1, ({Op1BaseType}*)(_dataTable.inArray1Ptr)),
+                ({Op2BaseType}*)_dataTable.inArray2Ptr,
+                {LoadIsa}.Load{Op3VectorType}(loadMask3, ({Op3BaseType}*)(_dataTable.inArray3Ptr))
+            );
+
+            Unsafe.Write(_dataTable.outArrayPtr, result);
+            ValidateResult(_dataTable.inArray1Ptr, _dataTable.inArray2Ptr, _dataTable.inArray3Ptr, _dataTable.outArrayPtr);
+        }
+
+        public void RunBasicScenario_LoadFirstFaulting()
+        {
+            TestLibrary.TestFramework.BeginScenario(nameof(RunBasicScenario_LoadFirstFaulting));
+
+            {Op1VectorType}<{Op1BaseType}> loadMask1 = Sve.CreateTrueMask{Op1BaseType}(SveMaskPattern.All);
+            {Op3VectorType}<{Op3BaseType}> loadMask3 = Sve.CreateTrueMask{Op3BaseType}(SveMaskPattern.All);
+
+            var op1 = {LoadIsa}.Load{Op1VectorType}(loadMask1, ({Op1BaseType}*)(_dataTable.inArray1Ptr));
+            ref var op2Ref = ref (_dataTable.inBounded.Span.GetPinnableReference());
+            var op3 = {LoadIsa}.Load{Op3VectorType}(loadMask3, ({Op3BaseType}*)(_dataTable.inArray3Ptr));
+
+            // When testing first-faulting behavior, we need to make sure we can get the first element.
+            // So set the first active element of the index vector to 0.
+            for (var i = 0; i < Vector<{Op3BaseType}>.Count; i++)
+            {
+                // op1 is the mask for GatherVector.
+                if (op1[i] != 0)
+                {
+                    op3 = op3.WithElement<{Op3BaseType}>(i, 0);
+                    break;
+                }
+            }
+
+            Sve.SetFfr(Sve.CreateTrueMaskByte(SveMaskPattern.All));
```

Comment:
`SetFfr` API actually maps to `wrffr` which is "[Write FFR](https://docsmirror.github.io/A64/2023-06/wrffr_f_p.html)"  and is different than "[setffr](https://docsmirror.github.io/A64/2023-06/setffr_f.html)"  which is actually used to initialize FFR to all true. So your intention is to actually use `setffr` instruction, but under the hood it is using `wrffr`, which should not be used like that as I pointed above. We should have `WriteFfr` API that maps to `wrffr` and the `SetFfr` should map to the `setffr` instruction.

Reply:
Edit: Alright, I just realized that `WriteFfr` is present in https://github.com/dotnet/runtime/issues/94004, but was not included in the list of https://github.com/dotnet/runtime/issues/99957. I have updated the list to include it. Also the generated files that @a74nh might not have taken into account for this. So yes, there is a `WriteFfR` that maps to `wrffr` instruction and `SetFfr` API that maps to `setffr` instruction.

Reply:
Another Edit: as we spoke offline, it is fine to do `SetFfr(TrueAll)` because that will convert to `setffr` vs. `wrffr`. just add a comment about it though.